src/plots/cartesian/layout_defaults.js

```diff
@@ -230,7 +230,13 @@ module.exports = function supplyLayoutDefaults(layoutIn, layoutOut, fullData) {
                 axHasImage[axName]
               ));
 
+        var axTemplate;
+        if(Lib.isPlainObject(layoutOut._template)) {
+            axTemplate = layoutOut._template[axLayoutOut._name];
```

Comment:
This isn't quite right - `Template.newContainer` also lets `xaxis` from the template be used for `xaxis2`, xaxis3` etc in the actual plot. But do you need `axTemplate` at all? Can't you just reference `containerOut._template` where you currently reference `options.axTemplate`?

Reply:
@alexcjohnson Good call. Fixed in https://github.com/plotly/plotly.js/pull/4670/commits/2494c4e1ad8718bcd4e23d559e3caa0509162a44.